src/plugins/discover/public/application/angular/doc_table/create_doc_table_embeddable.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { render } from 'react-dom';
+import React, { useRef, useEffect } from 'react';
+import { I18nProvider } from '@kbn/i18n/react';
+import { getServices } from '../../../kibana_services';
+import { AngularScope, DocTableLegacyProps, injectAngularElement } from './create_doc_table_react';
+
+export interface DocTableEmbeddableProps extends Partial<DocTableLegacyProps> {
+  refs: HTMLElement;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function getRenderFn(domNode: Element, props: any) {
+  const directive = {
+    template: `<doc-table
+        class="panel-content"
+        columns="renderProps.columns"
+        data-description="{{renderProps.searchDescription}}"
+        data-shared-item
+        data-test-subj="embeddedSavedSearchDocTable"
+        data-title="{{renderProps.sharedItemTitle}}"
+        filter="renderProps.onFilter"
+        hits="renderProps.rows"
+        index-pattern="renderProps.indexPattern"
+        is-loading="renderProps.isLoading"
+        on-add-column="renderProps.onAddColumn"
+        on-change-sort-order="renderProps.onSort"
+        on-move-column="renderProps.onMoveColumn"
+        on-remove-column="renderProps.onRemoveColumn"
+        render-complete
+        sorting="renderProps.sort"
+        total-hit-count="renderProps.totalHitCount"
+        use-new-fields-api="renderProps.useNewFieldsApi"</doc-table>`,
+  };
+
+  return async () => {
+    try {
+      const injector = await getServices().getEmbeddableInjector();
+      return await injectAngularElement(domNode, directive.template, props, injector);
+    } catch (e) {
+      render(<div>error</div>, domNode);
+    }
+  };
+}
+
+export function DiscoverDocTableEmbeddable(props: DocTableEmbeddableProps) {
+  return (
+    <I18nProvider>
+      <DocTableLegacyInner {...props} />
+    </I18nProvider>
+  );
+}
+
+function DocTableLegacyInner(renderProps: DocTableEmbeddableProps) {
+  const scope = useRef<AngularScope | undefined>();
+
+  useEffect(() => {
+    if (renderProps.refs) {
+      const fn = getRenderFn(renderProps.refs, renderProps);
+      fn().then((newScope) => {
+        scope.current = newScope;
+      });
+    }
```

Comment:
I think I've found the reason why the domNode count is much higher in this PR compared to the one in master. So currently, with this code, the classic table is re-rendered completely with every change. with the following change, it would just render once, and then angular would take care of the additional renderings:

```TypeScript
if (renderProps.refs && !scope.current) {
      const fn = getRenderFn(renderProps.refs, renderProps);
      fn().then((newScope) => {
        scope.current = newScope;
      });
} else if (scope && scope.current) {
      scope.current.renderProps = { ...renderProps };
      scope.current.$applyAsync();
    }
```
Some additional typings is necessary to satisfy TypeScript, best to look here how it was solved: 

https://github.com/elastic/kibana/blob/3ad0532cfb079ce270baf206de721d1f792db6a4/src/plugins/discover/public/application/angular/doc_table/create_doc_table_react.tsx#L128

Reply:
:art: Optional chaining :) 

```ts
ref?.current
scope?.current
```